### PR TITLE
Bugfix/rmq device

### LINKED ISSUE #28: Read and test cuda return values
We are not always looking at the return value of cuda api invocations. We should fix this.

### PR DESCRIPTION
**Pull request by koparasy**
Fixes #28 and makes corrects RMQ bug when invoking copying.

### CODE REVIEW

File: src/AMSWorkflow/ams_wf/AMSDBStage.py [hunk 1]
```diff
@@ -81,5 +83,5 @@ def main():
 
 
 if __name__ == "__main__":
-    print("{0}".format(" ".join(sys.argv)))
+    main()
```
Comment (lpottier) [inline src/AMSWorkflow/ams_wf/AMSDBStage.py:86]: We have two `main()` calls here?

File: src/AMSWorkflow/ams/stage.py [hunk 2]
```diff
@@ -220,6 +220,7 @@ def header_format(self) -> str:
         - 4 bytes are the number of elements in the message. Limit max: 2^32 - 1
         - 2 bytes are the input dimension. Limit max: 65535
         - 2 bytes are the output dimension. Limit max: 65535
+        - 4 bytes are for aligning memory to 8
```
Comment (lpottier) [inline src/AMSWorkflow/ams/stage.py:223]: If you want to align on 16 bits, we could just increase the size of a few fields (for ex. add two bits to input dim and output dim, or add two bits to the MPI ranks and one bit to the size of header and the precision.

**Review by lpottier — APPROVED** [review]
The PR looks very good, thanks.

File: src/AMSWorkflow/ams_wf/AMSDBStage.py (hunk 1 above)
Comment (koparasy) [reply src/AMSWorkflow/ams_wf/AMSDBStage.py:86]: Good catch. Thx

File: src/AMSWorkflow/ams/stage.py (hunk 2 above)
Comment (koparasy) [reply src/AMSWorkflow/ams/stage.py:223]: If that becomes an issue we can use those bytes for "free" on the required field.

**Pull request merged**